framework/framework.go

```diff
@@ -93,11 +94,38 @@ func New(config Config) (*Framework, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.ResourceRouter must not be empty")
 	}
 
+	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		ctx = canceledcontext.NewContext(ctx, make(chan struct{}))
+
+		if config.InitCtxFunc != nil {
+			var err error
+			ctx, err = config.InitCtxFunc(ctx, obj)
+			if err != nil {
+				return nil, microerror.Maskf(err, "initializing context")
+			}
+		}
+
+		accessor, err := meta.Accessor(obj)
```

Comment:
It would be cool if we can do it in such a general way here in operatorkit, but see my concerns in the other comment.